vnet/lib/vnet/models/network.rb

```diff
@@ -35,6 +42,13 @@ class Network < Base
     # 0011_assoc_interface
     interface_networks: :destroy
 
+    def before_destroy
+      # the association_dependencies plugin doesn't allow us to destroy because it's a many to many relation
+      self.ip_leases.each { |lease| lease.destroy }
```

Comment:
If we can't delete through many_to_many, wouldn't it be possible to add a assoc_dep for ip_leases in ip_addresses?

There are other models that can delete ip_leases through assoc_dep, like mac_leases.

If you look at mac_leases you will find it deletes _mac_address.